src/app/views/editors/ckeditor/ckeditor_upload_plugin.ts

```diff
@@ -1,26 +1,75 @@
+///<reference path="../upload/renuo_upload_html_generator.ts"/>
+///<reference path="../upload/renuo_upload.d.ts"/>
 class CkeditorUploadPlugin {
+  constructor(private domContentBlock:DomContentBlock) {
+  }
+
   registerPlugin(editor:CKEDITOR.editor) {
-    return;
-
-    // TODO: implement this
-    //editor.addCommand('renuoUpload', {
-    //  exec: function (editor) {
-    //    const now = new Date();
-    //    editor.insertHtml('The current date and time is: <em>' + now.toString() + '</em>');
-    //  }
-    //});
-
-    //const pluginName = 'renuoUpload';
-    //CKEDITOR.plugins.add(pluginName, {
-    //  init: function (editor) {
-    //    editor.addCommand(pluginName, startUpload);
-    //    editor.ui.addButton(pluginName, {
-    //      label: 'Renuo Upload',
-    //      // icon: 'renuo-upload.gif',
-    //      toolbar: 'insert',
-    //      command: pluginName
-    //    });
-    //  }
-    //});
+    if (!this.domContentBlock.hasRenuoUpload()) return true;
+
+    editor.addCommand('renuoUpload', {
+      exec: (editor) => {
+        this.openDropzone(editor);
+        return true;
+      }
+    });
+    editor.ui.addButton('uploadButton', {
+      label: 'File/Image Upload',
+      command: 'renuoUpload',
+      toolbar: 'insert',
+      icon: 'imagebutton'
+    });
   }
+
+  private closeDropzone():void {
+    jQuery('#dropzone-overlay').remove();
+  }
+
+  private openDropzone(editor:CKEDITOR.editor):void {
+    const dropzoneOverlay = jQuery(`<div id="dropzone-overlay"></div>`);
+    const dropzoneContainer = jQuery(`<div class="dropzone-container"></div>`);
+    const dropzone = jQuery('<div></div>');
+    const renuoUploadCredentials = this.domContentBlock.renuoUploadCredentials;
+    dropzone.addClass('renuo-upload dz-clickable').attr('data-apikey', renuoUploadCredentials.apiKey)
+      .attr('data-signingurl', renuoUploadCredentials.signingUrl);
+    dropzoneOverlay.append(dropzoneContainer);
+    dropzoneContainer.append(dropzone);
+    jQuery('body').append(dropzoneOverlay);
+    jQuery('<style type="text/css">' +
```

Comment:
i haven't run the code in my browser, but should we use a hand cursor for the background for closing?